src/dotnet/ShellShim/ShellShimRepository.cs

```diff
@@ -19,15 +19,14 @@ internal class ShellShimRepository : IShellShimRepository
     {
         private const string LauncherExeResourceName = "Microsoft.DotNet.Tools.Launcher.Executable";
         private const string LauncherConfigResourceName = "Microsoft.DotNet.Tools.Launcher.Config";
+        private readonly DirectoryPath _globalShimsDirectory;
 
-        private readonly DirectoryPath _shimsDirectory;
-
-        public ShellShimRepository(DirectoryPath shimsDirectory)
+        public ShellShimRepository(DirectoryPath globalShimsDirectory)
```

Comment:
Regarding my comment above re: the `IShellShimRepository` change, I'd really like to see this type not care about global/local and have that only be a concern of the install/uninstall commands (i.e. the path passed into the constructor determines local vs. global).